spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java

```diff
@@ -160,4 +165,19 @@ public Server armeriaServer(
         logger.info("Armeria server started at ports: {}", server.activePorts());
         return server;
     }
+
+    /**
+     * A user can configure a {@link Server} by providing an {@link ArmeriaServerConfigurator} bean.
+     */
+    @Bean
+    @ConditionalOnProperty("server.shutdown")
+    public ArmeriaServerConfigurator gracefulShutdownServerConfigurator(
+            @Value("${server.shutdown}") String shutdown,
+            @Value("${spring.lifecycle.timeout-per-shutdown-phase:30s}") Duration duration) {
+        return sb -> {
+            if (GRACEFUL_SHUTDOWN.equalsIgnoreCase(shutdown)) {
+                sb.gracefulShutdownTimeout(duration, duration);
+            }
```

Comment:
Could move the `if` condition to out side of lambda?

Reply:
Fixed fbf3274